src/rust/ide/src/controller/visualisation.rs

```diff
@@ -0,0 +1,166 @@
+//! Visualisation controller.
+//!
+//! Ths Visualisation Controller is Responsible identifying all the available visualisation natively
+//! embedded in IDE and available within the project's `visualisation` folder. The
+//! `Visualisation Controller` lives as long as the `Project Controller`.
+
+use crate::prelude::*;
+
+use std::rc::Rc;
+use enso_protocol::language_server;
+
+
+
+// =============
+// === Error ===
+// =============
+
+/// Enumeration of errors used in `Visualisation Controller`.
+#[derive(Debug,Fail)]
+#[allow(missing_docs)]
+pub enum VisualiserError {
+    #[fail(display = "Visualiser \"{}\" not found.", identifier)]
+    NotFound {
+        identifier : VisualiserIdentifier
+    }
+}
+
+
+
+// ===============================
+// === VisualisationIdentifier ===
+// ===============================
+
+/// This enum is used to identify a visualiser both in the project folder or natively embedded in
+/// IDE.
+#[derive(Debug,Clone,PartialEq,Eq,Display)]
+#[allow(missing_docs)]
+pub enum VisualiserIdentifier {
+    Embedded(String),
+    File(language_server::Path)
+}
+
+
+
+// ==============
+// === Handle ===
+// ==============
+
+/// Visualisation Controller's state.
+#[derive(Debug,Clone)]
+pub struct Handle {
+    language_server_rpc  : Rc<language_server::Connection>,
+    embedded_visualisers : HashMap<String,String>
+}
+
+impl Handle {
+    /// Creates a new visualisation controller.
+    pub fn new
+    ( language_server_rpc  : Rc<language_server::Connection>
+    , embedded_visualisers : HashMap<String,String>) -> Self {
+        Self {language_server_rpc,embedded_visualisers}
+    }
+
+    /// Get a list of all available visualisers.
+    pub async fn list_visualisers(&self) -> FallibleResult<Vec<VisualiserIdentifier>> {
+        let root_id    = self.language_server_rpc.content_root();
+        let vis_path   = language_server::Path{root_id,segments:vec!["visualisation".into()]};
```

Comment:
The path to the visualizations in project could be a spearate function, and definitely the "visualisation" string should be a separate constant.

Reply:
Yes, please, be sure it never ever happens in PRs . No hardcoded literals. Ever! :)